gtests/net/packetdrill/packet.c

```diff
@@ -267,10 +268,79 @@ struct header_type_info *header_type_info(enum header_t header_type)
 	return &header_types[header_type];
 }
 
+static struct tcp_option *find_dss_option(struct packet *packet, char **error)
+{
+	struct tcp_options_iterator iter;
+	struct tcp_option *opt;
+
+	opt = tcp_options_begin(packet, &iter);
+	while (opt != NULL) {
+		if (opt->kind == TCPOPT_MPTCP &&
+		    opt->data.dss.subtype == DSS_SUBTYPE &&
+		    opt->data.dss.flag_M == 1)
+			return opt;
+
+		opt = tcp_options_next(&iter, error);
+	}
+	return NULL;
+}
+
+static void aggregate_tcp_options(struct packet *packet,
+				  struct packet *first_packet,
+				  char **error)
+{
+	struct tcp_option *d_opt, *s_opt;
+	struct dsn *d_dsn, *s_dsn;
+	u16 delta;
+
+	d_opt = find_dss_option(packet, error);
+	s_opt = find_dss_option(first_packet, error);
+
+	if (!d_opt && !s_opt)
+		return;
+
+	if (!d_opt || !s_opt ||
```

Comment:
just wondering: could you have this case if a fallback happened? → one of the two (`d_opt`, `s_opt`) is NULL, but not the other one?

Reply:
on fallback or any other non aggregation scenario we should error out, as the current code should do.